src/plugins/intel_gna/src/transformations/ts_concat.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/ts_concat.hpp"
+
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "backend/gna_limitations.hpp"
+#include "common/graph_utils.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset12.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/transformation_helper.hpp"
+
+using namespace ov;
+using namespace ov::opset12;
+using namespace ov::pass::pattern;
+using namespace ov::intel_gna;
+using namespace ov::intel_gna::pass;
+using namespace ov::intel_gna::pass::helper;
+using namespace ov::intel_gna::limitations;
+
+namespace {
+
+bool is_concat_sinked(const Output<Node>& output) {
+    auto concat_node = ov::as_type_ptr<Concat>(output.get_node_shared_ptr());
+
+    const Shape concat_output_shape = concat_node->get_output_shape(0);
+    const int64_t axis = concat_node->get_concatenation_axis();
+    if (graph_utils::get_first_valuable_dim_id(concat_output_shape) != axis)
```

Comment:
Could you provide an examples?

Reply:
That is a different level of the checking, for this transformation it is important to check that axis is not supported by default. I know only one case when it can be supported - when Concat inputs connected directly to network inputs, but this is not a pattern of this transformation.